service/src/main/java/uk/gov/hmcts/reform/fpl/service/legalcounsel/ManageLegalCounselService.java

```diff
@@ -0,0 +1,169 @@
+import uk.gov.hmcts.reform.fpl.model.Respondent;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.model.event.ManageLegalCounselEventData;
+import uk.gov.hmcts.reform.fpl.model.interfaces.WithSolicitor;
+import uk.gov.hmcts.reform.fpl.service.CaseConverter;
+import uk.gov.hmcts.reform.fpl.service.CaseRoleLookupService;
+import uk.gov.hmcts.reform.fpl.service.EventService;
+import uk.gov.hmcts.reform.fpl.service.OrganisationService;
+import uk.gov.hmcts.reform.rd.model.Organisation;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static java.lang.String.format;
+import static java.util.Arrays.asList;
+import static java.util.Collections.emptyList;
+import static java.util.function.Predicate.not;
+import static uk.gov.hmcts.reform.fpl.enums.SolicitorRole.Representing.CHILD;
+import static uk.gov.hmcts.reform.fpl.enums.SolicitorRole.Representing.RESPONDENT;
+
+@RequiredArgsConstructor
+@Component
+public class ManageLegalCounselService {
+
+    private static final List<SolicitorRole.Representing> RELEVANT_REPRESENTED_PARTY_TYPES = asList(RESPONDENT, CHILD);
+
+    private static final String UNREGISTERED_USER_ERROR_MESSAGE_TEMPLATE = "Unable to grant access "
+        + "[%s is not a Registered User] - Email address for Legal representative is not registered on the system. "
+        + "They can register at https://manage-org.platform.hmcts.net/register-org/register";
+
+    private final CaseConverter caseConverter;
+    private final CaseRoleLookupService caseRoleLookupService;
+    private final OrganisationService organisationService;
+    private final EventService eventPublisher;
+
+    public List<Element<LegalCounsellor>> retrieveLegalCounselForLoggedInSolicitor(CaseDetails caseDetails) {
+        String caseId = caseDetails.getId().toString();
+        List<SolicitorRole> caseSolicitorRoles = caseRoleLookupService.getCaseSolicitorRolesForCurrentUser(caseId);
+
+        CaseData caseData = caseConverter.convert(caseDetails);
+        return caseSolicitorRoles.stream()
+            .filter(solicitorRole -> RELEVANT_REPRESENTED_PARTY_TYPES.contains(solicitorRole.getRepresenting()))
+            .findFirst()
+            .map(solicitorRole -> {
+                int representedPartyIndex = solicitorRole.getIndex();
+                return solicitorRole.getRepresenting().getTarget().apply(caseData).get(representedPartyIndex);
+            })
+            .map(Element::getValue)
+            .map(WithSolicitor::getLegalCounsellors)
+            .orElse(emptyList());
+    }
+
+    public void updateLegalCounsel(CaseDetails caseDetails) {
+        String caseId = caseDetails.getId().toString();
+        List<SolicitorRole> caseSolicitorRoles = caseRoleLookupService.getCaseSolicitorRolesForCurrentUser(caseId);
+
+        CaseData caseData = caseConverter.convert(caseDetails);
```

Comment:
vv minor but if this is the first line of the method then the id can be extracted from case data using `caseData.getId()`

Reply:
Got you. Cool.
Will do that.

Reply:
Done!